models/issue_watch.go

```diff
@@ -21,7 +21,21 @@ type IssueWatchList []*IssueWatch
 
 // CreateOrUpdateIssueWatch set watching for a user and issue
 func CreateOrUpdateIssueWatch(userID, issueID int64, isWatching bool) error {
-	iw, exists, err := getIssueWatch(x, userID, issueID)
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
```

Comment:
Why do we need a separate function? As long as `createOrUpdateIssueWatch()` performs only one statement (either insert or update), IMHO there no need for a transaction here. Unless you need to call `createOrUpdateIssueWatch()` as part of a bigger transaction from other parts of the code.

Reply:
ok ... I'll revert c1de540